gpgme-sharp/Context.cs

```diff
@@ -142,6 +142,53 @@ private void Initialize(ref ContextSigners signers, ref ContextSignatureNotation
                 }
             }
         }
+
+        public PinentryMode PinentryMode
+        {
+            get
+            {
+                if (IsValid)
+                {
+                    lock (CtxLock)
+                    {
+                        var mode = libgpgme.NativeMethods.gpgme_get_pinentry_mode(CtxPtr);
+                        return (PinentryMode)mode;
+                    }
+                }
+                throw new InvalidContextException();
+            }
+            set
+            {
+                if (IsValid)
+                {
+                    lock (CtxLock)
+                    {
+                        var mode = (gpgme_pinentry_mode_t)value;
+                        int err = libgpgme.NativeMethods.gpgme_set_pinentry_mode(CtxPtr, mode);
+
+                        gpg_err_code_t errcode = libgpgme.gpgme_err_code(err);

```

Comment:
I'll submit a separate PR to clean up all this error handling code - A lot of it is copy-pasted right now, unfortunately.